### PR TITLE
Task-37857: Implement dlp-quarantine administration portlet

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

**Comment by ahamdi** [comment]
@azayati @AzmiTouil Why are latest PRs on Develop ?

File: apps/pom.xml [hunk 1]
```diff
@@ -21,5 +21,6 @@
     <module>portlet-documents</module>
     <module>portlet-clouddrives</module>
     <module>portlet-transferrules</module>
+    <module>portlet-quarantine</module>
```
Comment (azayati) [inline apps/pom.xml]:
```suggestion
    <module>portlet-dlp-quarantine</module>
```

File: apps/portlet-quarantine/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "quarantine",
```
Comment (azayati) [inline apps/portlet-quarantine/package.json]:
```suggestion
  "name": "dlp-quarantine",
```

File: apps/portlet-quarantine/package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  "version": "1.0.0",
+  "description": "eXo Quarantine",
```
Comment (azayati) [inline apps/portlet-quarantine/package.json]:
```suggestion
  "description": "eXo Dlp Quarantine",
```

File: apps/portlet-quarantine/src/main/resources/locale/portlet/quarantine_fr.properties [hunk 4]
```diff
@@ -0,0 +1 @@
+documents.quarantine.label=Quarantaine
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/resources/locale/portlet/quarantine_fr.properties]:
```suggestion
documents.dlp.quarantine.label=Quarantaine
```
Put the other labels

File: apps/portlet-quarantine/package.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/exoplatform/ecms"
+  },
+  "scripts": {
+    "start": "webpack --config webpack.dev.js --mode development",
+    "dev": "webpack-dev-server --config webpack.dev.js --mode development --hot",
+    "watch": "webpack --config webpack.dev.js --progress --colors --watch -d",
+    "lint": "eslint --fix ./src/main/webapp/vue-apps/quarantine/components/QuarantineApp.vue ./src/main/webapp/vue-app/quarantine/*.js",
```
Comment (azayati) [inline apps/portlet-quarantine/package.json]:
```suggestion
    "lint": "eslint --fix ./src/main/webapp/vue-apps/dlp-quarantine/components/DlpQuarantineApp.vue ./src/main/webapp/vue-app/dlp-quarantine/*.js",
```

File: apps/portlet-quarantine/pom.xml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>ecms-apps</artifactId>
+    <groupId>org.exoplatform.ecms</groupId>
+    <version>6.2.x-SNAPSHOT</version>
+  </parent>
+  <artifactId>ecms-apps-portlet-quarantine</artifactId>
```
Comment (azayati) [inline apps/portlet-quarantine/pom.xml]:
```suggestion
  <artifactId>ecms-apps-portlet-dlp-quarantine</artifactId>
```

File: apps/portlet-quarantine/pom.xml [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  <packaging>war</packaging>
+  <name>eXo PLF:: Quarantine Admin Portlet</name>
```
Comment (azayati) [inline apps/portlet-quarantine/pom.xml]:
```suggestion
  <name>eXo PLF:: Dlp Quarantine Admin Portlet</name>
```

File: apps/portlet-quarantine/pom.xml [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  <description>eXo DMS Portlet Java Content: Quarantine Admin</description>
```
Comment (azayati) [inline apps/portlet-quarantine/pom.xml]:
```suggestion
  <description>eXo DMS Portlet Java Content: Dlp Quarantine Admin</description>
```

File: apps/portlet-quarantine/pom.xml [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+  <build>
+    <finalName>quarantine</finalName>
```
Comment (azayati) [inline apps/portlet-quarantine/pom.xml]:
```suggestion
    <finalName>dlp-quarantine</finalName>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/configuration.xml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.exoplatform.org/xml/ns/kernel_1_3.xsd http://www.exoplatform.org/xml/ns/kernel_1_3.xsd"
+  xmlns="http://www.exoplatform.org/xml/ns/kernel_1_3.xsd">
+
+  <import>war:/conf/quarantine/portal/portal-configuration.xml</import>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/configuration.xml]:
```suggestion
  <import>war:/conf/dlp-quarantine/portal/portal-configuration.xml</import>
```

File: apps/portlet-quarantine/src/main/resources/locale/navigation/group/platform/administrators_en.properties [hunk 11]
```diff
@@ -0,0 +1 @@
+administration.quarantine.label=Quarantine
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/resources/locale/navigation/group/platform/administrators_en.properties]:
```suggestion
administration.dlp.quarantine.label=Quarantine
```

File: apps/portlet-quarantine/src/main/resources/locale/navigation/group/platform/administrators_fr.properties [hunk 12]
```diff
@@ -0,0 +1 @@
+administration.quarantine.label=Quarantaine
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/resources/locale/navigation/group/platform/administrators_fr.properties]:
```suggestion
administration.dlp.quarantine.label=Quarantaine
```

File: apps/portlet-quarantine/src/main/resources/locale/portlet/quarantine_en.properties [hunk 13]
```diff
@@ -0,0 +1,10 @@
+documents.quarantine.label=Quarantine
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/resources/locale/portlet/quarantine_en.properties]:
```suggestion
documents.dlp.quarantine.label=Quarantine
```
same for other labels

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+    <title>Quarantine</title>
+    <access-permissions>member:/platform/administrators</access-permissions>
+    <edit-permission>manager:/platform/administrators</edit-permission>
+    <show-max-window>false</show-max-window>
+    <portlet-application>
+      <portlet>
+        <application-ref>quarantine</application-ref>
+        <portlet-ref>quarantinePortlet</portlet-ref>
+      </portlet>
+      <title>Quarantine Administration</title>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml]:
```suggestion
      <title>Dlp Quarantine Administration</title>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/portlet.xml [hunk 15]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<portlet-app xmlns="http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd" version="2.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd
+   http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd">
+  <portlet>
+    <portlet-name>quarantinePortlet</portlet-name>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
    <portlet-name>dlpQuarantinePortlet</portlet-name>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/portlet.xml [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+    <display-name xml:lang="EN">Quarantine portlet</display-name>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
    <display-name xml:lang="EN">Dlp Quarantine portlet</display-name>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/portlet.xml [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+      <name>exo.profiles</name>
+      <value>ecms</value>
+    </init-param>
+    <expiration-cache>-1</expiration-cache>
+    <cache-scope>PUBLIC</cache-scope>
+    <supports>
+      <mime-type>text/html</mime-type>
+    </supports>
+    <portlet-info>
+      <title>Quarantine portlet for eXo Platform</title>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
      <title>Dlp Quarantine portlet for eXo Platform</title>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/gatein-resources.xml [hunk 18]
```diff
@@ -0,0 +1,28 @@
+<gatein-resources xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.exoplatform.org/xml/ns/gatein_resources_1_4 http://www.exoplatform.org/xml/ns/gatein_resources_1_4"
+  xmlns="http://www.exoplatform.org/xml/ns/gatein_resources_1_4">
+
+  <module>
+    <name>quarantine</name>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/gatein-resources.xml]:
```suggestion
    <name>dlp-quarantine</name>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/navigation.xml [hunk 19]
```diff
@@ -0,0 +1,14 @@
+<node-navigation
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.gatein.org/xml/ns/gatein_objects_1_3 http://www.gatein.org/xml/ns/gatein_objects_1_3"
+        xmlns="http://www.gatein.org/xml/ns/gatein_objects_1_3">
+  <priority>1</priority>
+  <page-nodes>
+     <node>
+      <uri>quarantine</uri>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/navigation.xml]:
```suggestion
      <uri>dlp-quarantine</uri>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/navigation.xml [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+      <name>quarantine</name>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/navigation.xml]:
```suggestion
      <name>dlp-quarantine</name>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/navigation.xml [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+      <label>#{administration.quarantine.label}</label>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/navigation.xml]:
```suggestion
      <label>#{administration.dlp.quarantine.label}</label>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/navigation.xml [hunk 22]
```diff
@@ -0,0 +1,14 @@
...
+      <page-reference>group::/platform/administrators::quarantine</page-reference>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/navigation.xml]:
```suggestion
      <page-reference>group::/platform/administrators::dlp-quarantine</page-reference>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+    License along with this software; if not, write to the Free
+    Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+    02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+<page-set
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.gatein.org/xml/ns/gatein_objects_1_3 http://www.gatein.org/xml/ns/gatein_objects_1_3"
+        xmlns="http://www.gatein.org/xml/ns/gatein_objects_1_3">
+  <page>
+    <name>quarantine</name>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml]:
```suggestion
    <name>dlp-quarantine</name>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml [hunk 24]
```diff
@@ -0,0 +1,37 @@
...
+    <title>Quarantine</title>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml]:
```suggestion
    <title>Dlp Quarantine</title>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml [hunk 25]
```diff
@@ -0,0 +1,37 @@
...
+    <access-permissions>member:/platform/administrators</access-permissions>
+    <edit-permission>manager:/platform/administrators</edit-permission>
+    <show-max-window>false</show-max-window>
+    <portlet-application>
+      <portlet>
+        <application-ref>quarantine</application-ref>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml]:
```suggestion
        <application-ref>dlp-quarantine</application-ref>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+        <portlet-ref>quarantinePortlet</portlet-ref>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml]:
```suggestion
        <portlet-ref>dlpQuarantinePortlet</portlet-ref>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml (hunk 14 above)
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/conf/quarantine/portal/group/platform/administrators/pages.xml]:
```suggestion
      <title>Dlp Quarantine Administration</title>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/gatein-resources.xml [hunk 27]
```diff
@@ -0,0 +1,28 @@
...
+    <script>
+      <minify>false</minify>
+      <path>/javascript/quarantine.bundle.js</path>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/gatein-resources.xml]:
```suggestion
      <path>/javascript/dlpQuarantine.bundle.js</path>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/portlet.xml [hunk 28]
```diff
@@ -0,0 +1,30 @@
...
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd
+   http://java.sun.com/xml/ns/portlet/portlet-app_2_0.xsd">
+  <portlet>
+    <portlet-name>quarantinePortlet</portlet-name>
+    <display-name xml:lang="EN">Quarantine portlet</display-name>
+    <portlet-class>org.exoplatform.commons.api.portlet.GenericDispatchedViewPortlet</portlet-class>
+    <init-param>
+      <name>portlet-view-dispatched-file-path</name>
+      <value>/quarantine.html</value>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/portlet.xml]:
```suggestion
      <value>/dlpQuarantine.html</value>
```

File: apps/portlet-quarantine/src/main/webapp/WEB-INF/web.xml [hunk 29]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<web-app
+        version="3.0"
+        metadata-complete="true"
+        xmlns="http://java.sun.com/xml/ns/javaee"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd">
+  <display-name>quarantine</display-name>
```
Comment (azayati) [inline apps/portlet-quarantine/src/main/webapp/WEB-INF/web.xml]:
```suggestion
  <display-name>dlp-quarantine</display-name>
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Review by azayati — DISMISSED** [review]
LGTM after QA validation

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**